bundles/org.eclipse.osgi.services/META-INF/MANIFEST.MF

```diff
@@ -2,56 +2,28 @@ Manifest-Version: 1.0
+Bundle-Version: 3.11.000.qualifier
 Bundle-Description: %osgiServicesDes
 Bundle-Localization: plugin
 Bundle-Vendor: %eclipse.org
 Bundle-DocUrl: http://www.eclipse.org
 Bundle-ContactAddress: www.eclipse.org
-Export-Package: org.osgi.service.cm;version="1.6";uses:="org.osgi.framework",
- org.osgi.service.component;version="1.4";uses:="org.osgi.framework",
- org.osgi.service.component.annotations;version="1.3",
- org.osgi.service.component.runtime;version="1.4";uses:="org.osgi.framework,org.osgi.util.promise,org.osgi.service.component.runtime.dto",
- org.osgi.service.component.runtime.dto;version="1.4";uses:="org.osgi.dto,org.osgi.framework.dto",
- org.osgi.service.device;version="1.1";uses:="org.osgi.framework",
- org.osgi.service.event;version="1.4";uses:="org.osgi.framework",
- org.osgi.service.http;version="1.2.1";uses:="javax.servlet,javax.servlet.http",
- org.osgi.service.http.context;version="1.1";uses:="org.osgi.framework,javax.servlet.http",
- org.osgi.service.http.runtime;version="1.1";uses:="org.osgi.service.http.runtime.dto",
- org.osgi.service.http.runtime.dto;version="1.1";uses:="org.osgi.dto,org.osgi.framework.dto",
- org.osgi.service.http.whiteboard;version="1.1",
- org.osgi.service.log;version="1.5";uses:="org.osgi.framework",
- org.osgi.service.metatype;version="1.4";uses:="org.osgi.framework",
- org.osgi.service.provisioning;version="1.2",
- org.osgi.service.upnp;version="1.2",
- org.osgi.service.useradmin;version="1.1";uses:="org.osgi.framework",
- org.osgi.service.wireadmin;version="1.0.1";uses:="org.osgi.framework"
-Import-Package: javax.servlet;resolution:=optional,
- javax.servlet.http;resolution:=optional,
- org.osgi.dto;version="1.0",
- org.osgi.framework;version="1.6",
- org.osgi.framework.dto;version="1.8.0",
- org.osgi.service.cm;version="[1.6,1.7)",
- org.osgi.service.component;version="[1.4,1.5)",
- org.osgi.service.component.annotations;version="[1.3,1.4)",
- org.osgi.service.component.runtime; version="[1.4,1.5)",
- org.osgi.service.component.runtime.dto; version="[1.4,1.5)",
- org.osgi.service.device;version="[1.1,1.2)",
- org.osgi.service.event;version="[1.4,1.5)",
- org.osgi.service.http;version="[1.2,1.3)",
- org.osgi.service.http.context;version="[1.1,1.2)",
- org.osgi.service.http.runtime;version="[1.1,1.2)",
- org.osgi.service.http.runtime.dto;version="[1.1,1.2)",
- org.osgi.service.http.whiteboard;version="[1.1,1.2)",
- org.osgi.service.log;version="[1.5,1.6)",
- org.osgi.service.metatype;version="[1.4,1.5)",
- org.osgi.service.provisioning;version="[1.2,1.3)",
- org.osgi.service.upnp;version="[1.2,1.3)",
- org.osgi.service.useradmin;version="[1.1,1.2)",
- org.osgi.service.wireadmin;version="[1.0,1.1)",
- org.osgi.util.function;version="[1.0,2.0)",
- org.osgi.util.promise;version="[1.0,2.0)"
+Export-Package: org.osgi.service.component.annotations;version="1.3"
+Import-Package: org.osgi.annotation.versioning;version="[1.1.2,1.2.0)",
+ org.osgi.service.component.annotations;version="[1.3,1.4)"
```

Comment:
I imagine the `versioning` package is imported here to allow the build to succeed for the remaining source in this bundle.  Before we put some project JARs on the project classpath to do this.  We could make this import on `versioning` optional or go back to a project JAR or some other solution.

Reply:
But it is plainly wrong to have an Import-Package in the resulting binary just to make sure source code can compile. This is an example of how PDE conflates build configuration and build output in the Manifest.

Reply:
Restoring use of the osgi.annotation binary solves the need for the Import-Package but there should be some other way to get osgi.annotation on the compile path.

Reply:
Isn’t there an extra classpath property you can put in the build.properties for such build/compile time dependencies?  I think Tycho respects that too.

Reply:
There is `jars.extra.classpath = lib/osgi.annotation.jar` but this now refers to a local binary in the project source. It would be better if it could refer to some artifact in a repository like maven central.

Reply:
I think this is fine to have as an incremental improvement.  We wouldn't need any of this if we got rid of the org.eclipse.osgi.services bundle altogether.  But that takes a longer announcement and deprecation process and even then we will likely get loud objections to the removal.  The other option is to simply have the class files sucked into the produced JAR during the build from the available maven artifacts.  The disadvantage of that is that this will not work for the PDE project imported into the workspace.

Reply:
I agree with everything said in this thread. To me it looks like the root of this problem is that PDE does not support different _scopes_ of dependencies, for example like Maven does. I don't think this is a problem of OSGi because is 'just' the runtime and its PDEs job to assemble a suitable runtime respectively compile-time reactor/container.
PDE uses the same set of dependencies when resolving the PDE-State (the 'Compile-Container') and when launching an Eclipse-Application (the 'Runtime-Container'). But it could use different sets (e.g. compile-time only like for the annotations). To avoid the need ot add the `org.osgi.service.component.annotations` when using OSGi DS those annotations are treated specially by PDE. If one enables a corresponding setting the annotation-jar with desired version is added implicitly to the classpath at compile-time. So basically PDE already has different Sets of dependencies but in for a very special scenario. If this concept where generalized it could also avoid the need to embed the service.component.annotation jars in PDE.
There could also be another scope for _tests_ (about which I recently had a chat with @laeubi).

However this would be a completely different story (but I think an interesting one) and way out of scope of this change.
Therefore I would like to follow Tom's suggestion and proceed for now with the smaller steps we can do now :) ...  if the updated SDK-prereq-TP is published by the platform.aggr build, which fails due to a subsequent change.